### ISSUE TITLE
QuickFixJTemplate not found in spring context

### ISSUE THREAD

**Issue by ggershaw** [esanchezros/quickfixj-spring-boot-starter#97]
Hi,

I am trying to create a client that responds to the logon message with a market data request. For some reason, no matter how I try, the template is not found in the context.

I'm assuming that you allow the client to send messages with the template although I don't see an example of this in your repo. I could have missed it.

My code is here https://github.com/ggershaw/Examples. The branch is feature/rollback. The code that exercises your code is in the child module spring-boot-starter.

I tried to make my yml file closely resemble yours. I'm wondering if the template is a conditional bean and I am missing the config that brings it to life.

Would you mind creating an example of the client publishing a message using the template?

Thanks much

**Comment by esanchezros** [comment]
Hi @ggershaw 

Have a look at this example https://github.com/esanchezros/quickfixj-spring-boot-starter-examples/tree/master/simple-client-and-server, the `QuickFixJTemplate` is autowired in the `RestController`

**Comment by esanchezros** [comment]
The example is both a client and server. I see where the problem is, `LoginHandler` needs to be a `@Component` or `@Service` in order for Spring to autowire the dependency

**Comment by esanchezros** [comment]
I tried autowiring the `QuickFixJTemplate` bean on both client and server examples and it works. I wonder if Spring is trying to match the bean by name rather than type. Could you try renaming the constructor variable to `quickFixJTemplate`? I probably should check out the code and try debugging the issue

**Comment by esanchezros** [comment]
Hi @ggershaw 

It seems like the boot starter isn't working properly with Spring 3.0.0. Downgrading to 2.7.6 works as expected.

I have made it work with 3.0.0 by including this in `Config`:

```
@Configuration
@ImportAutoConfiguration(classes = {QuickFixJTemplateAutoConfiguration.class, QuickFixJClientAutoConfiguration.class})
public class Config {

}
```

I'll keep this ticket opened for a while and will create a separate ticket for this bug.

Thanks

**Comment by esanchezros** [comment]
If you downgrade to 2.7.6 it should start the initiator and will try connecting to the acceptor. If you want to continue using 3.0.0 then you need to use the workaround as stated before:

```
@Configuration @ImportAutoConfiguration(classes = {QuickFixJTemplateAutoConfiguration.class, QuickFixJClientAutoConfiguration.class}) public class Config { }
```

**Comment by esanchezros** [comment]
Fixed on 2.15.2
